btsieve/src/route_factory.rs

```diff
@@ -108,6 +108,17 @@ where
     let query_repository = warp::any().map(move || Arc::clone(&query_repository));
     let query_result_repository = warp::any().map(move || Arc::clone(&query_result_repository));
 
+    // validate query id function
+    let validate_query_id = warp::path::param::<String>().and_then(|id: String| {
```

Comment:
Since the validation is only doing a single check perhaps we could name this `validate_query_id_length`.  `validate_query_id` implies, to me, more extensive validation.  Especially considering the matching error is `QueryIdTooLong`.